diff -Naur pyglossary-4.0.11-orig/pyglossary/plugins/ebook_kobo.py pyglossary-4.0.11/pyglossary/plugins/ebook_kobo.py
--- pyglossary-4.0.11-orig/pyglossary/plugins/ebook_kobo.py	2020-12-23 18:41:43.000000000 -0800
+++ pyglossary-4.0.11/pyglossary/plugins/ebook_kobo.py	2021-02-14 23:38:09.509047305 -0800
@@ -85,7 +85,7 @@
 	WORDS_FILE_NAME = "words"
 
 	depends = {
-		"marisa_trie": "marisa-trie",
+		"marisa": "marisa",
 	}
 
 	def __init__(self, glos, **kwargs):
@@ -98,9 +98,9 @@
 		)
 		# img tag has no closing
 		try:
-			import marisa_trie
+			import marisa
 		except ModuleNotFoundError as e:
-			e.msg += f", run `{pip} install marisa-trie` to install"
+			e.msg += f", run `{pip} install marisa` to install"
 			raise e
 
 	def get_prefix(self, word: str) -> str:
@@ -236,8 +236,8 @@
 			yield from self.write_groups()
 
 	def finish(self) -> None:
-		import marisa_trie
+		import marisa
 		with indir(self._filename, create=False):
-			trie = marisa_trie.Trie(self._words)
+			trie = marisa.Trie(self._words)
 			trie.save(self.WORDS_FILE_NAME)
 		self._filename = None
diff -Naur pyglossary-4.0.11-orig/setup.py pyglossary-4.0.11/setup.py
--- pyglossary-4.0.11-orig/setup.py	2020-12-23 18:41:43.000000000 -0800
+++ pyglossary-4.0.11/setup.py	2021-02-14 23:37:20.660789530 -0800
@@ -126,7 +126,7 @@
 			"beautifulsoup4",
 			"PyICU",
 			"PyYAML",
-			"marisa-trie",
+			"marisa",
 			"libzim",
 			"python-lzo",
 		],
