diff -Naur btsync-gui-orig/btsyncagent.py btsync-gui/btsyncagent.py
--- btsync-gui-orig/btsyncagent.py	2015-05-10 22:02:37.375188029 +0100
+++ btsync-gui/btsyncagent.py	2015-05-10 22:06:52.979512620 +0100
@@ -49,7 +49,7 @@
 class BtSyncAgent(BtSyncApi):
 	# still hardcoded - this is the binary location of btsync when installing
 	# the package btsync-common
-	BINARY = '/usr/lib/btsync-common/btsync-core'
+	BINARY = '/usr/bin/btsync'
 
 	def __init__(self,args):
 		BtSyncApi.__init__(self)
@@ -154,7 +154,7 @@
 					logging.info ('Found running btsync agent. Stopping...')
 					os.kill (self.pid, signal.SIGTERM)
 					time.sleep(1)
-					
+
 				if not self.is_paused():
 					logging.info ('Starting btsync agent...')
 					self.start_agent()
@@ -348,7 +348,7 @@
 	@staticmethod
 	def get_api_key():
 		try:
-			akf = open('/usr/lib/btsync-gui/btsync-gui.key','r')
+			akf = open('/usr/share/btsync-gui/btsync-gui.key','r')
 			key = akf.readline()
 			akf.close()
 			return key.rstrip('\n\r')
