From 3ddb9865e49f81bef3a99417e4814ed59efaff5d Mon Sep 17 00:00:00 2001
From: Scott Marks <scott.marks@ka.je>
Date: Sat, 10 Feb 2024 01:10:29 -0500
Subject: [PATCH] #include <csdtint> to define uint8_t early

---
 Common/DtaOptions.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/Common/DtaOptions.h b/Common/DtaOptions.h
index c012af1d..4d885391 100644
--- a/Common/DtaOptions.h
+++ b/Common/DtaOptions.h
@@ -20,6 +20,7 @@ along with sedutil.  If not, see <http://www.gnu.org/licenses/>.
 
 #ifndef _DTAOPTIONS_H
 #define	_DTAOPTIONS_H
+#include <cstdint>
 
 /** Output modes */
 typedef enum _sedutiloutput {
@@ -48,7 +49,7 @@ typedef struct _DTA_OPTIONS {
 /** Print a usage message */
 void usage();
 /** Parse the command line and return a structure that describes the action desired
- * @param argc program argc parameter 
+ * @param argc program argc parameter
  * @param argv program argv paramater
  * @param opts pointer to options structure to be filled out
  */
@@ -126,6 +127,6 @@ i++;
 
 /** set the argc value for this parameter in the options structure */
 #define OPTION_IS(option_field) \
-				opts->option_field = ++i; 
+				opts->option_field = ++i;
 
 #endif /* _DTAOPTIONS_H */
