diff -Naur casync-2-orig/src/caencoder.c casync-2/src/caencoder.c
--- casync-2-orig/src/caencoder.c	2017-07-26 02:55:38.000000000 -0700
+++ casync-2/src/caencoder.c	2017-11-19 19:15:11.717250817 -0800
@@ -264,10 +264,12 @@
 
         n->fcaps = mfree(n->fcaps);
 
+#if HAVE_SELINUX
         if (n->selinux_label) {
                 freecon(n->selinux_label);
                 n->selinux_label = NULL;
         }
+#endif
 
         n->device_size = UINT64_MAX;
 
@@ -658,18 +660,20 @@
                 CaEncoder *e,
                 CaEncoderNode *n) {
 
+#if HAVE_SELINUX
         char *label;
         int r;
+#endif
 
         assert(e);
         assert(n);
 
         if ((e->feature_flags & CA_FORMAT_WITH_SELINUX) == 0)
                 return 0;
+#if HAVE_SELINUX
         if (n->selinux_label_valid)
                 return 0;
 
-#if HAVE_SELINUX
         if (n->fd >= 0)
                 r = fgetfilecon(n->fd, &label) < 0 ? -errno : 0;
         else {
@@ -706,10 +710,13 @@
 
                 n->selinux_label = label;
         }
-#endif
 
         n->selinux_label_valid = true;
         return 0;
+
+#else
+        return -EOPNOTSUPP;
+#endif
 }
 
 static int compare_xattr(const void *a, const void *b) {
