diff -Naur reddio-v0.46-orig/reddio reddio-v0.46/reddio
--- reddio-v0.46-orig/reddio	2021-08-10 05:16:32.000000000 -0700
+++ reddio-v0.46/reddio	2022-01-22 16:19:30.108566702 -0800
@@ -324,8 +324,11 @@
 {
 	$logged_in || return 1
 
+	_now=$(date +%s)
+	_now=${_now%.*}
+
 	# Refresh our token if necessary
-	if [ "$(date +%s)" -ge "$token_expires" ]; then
+	if [ "$_now" -ge "$token_expires" ]; then
 		info 'refreshing expired access token'
 
 		. "$lib_dir"/refresh-token.sh
diff -Naur reddio-v0.46-orig/share/reddio/cmd-print.sh reddio-v0.46/share/reddio/cmd-print.sh
--- reddio-v0.46-orig/share/reddio/cmd-print.sh	2021-08-10 05:16:32.000000000 -0700
+++ reddio-v0.46/share/reddio/cmd-print.sh	2022-01-22 18:00:47.158868423 -0800
@@ -170,6 +170,7 @@
 	fi
 
 	_now=$(date +%s)
+	_now=${_now%.*}
 }
 
 cmd_print()
@@ -400,9 +401,11 @@
 			*) score_plural=s ;;
 		esac
 
-		created_pretty=$(pretty_time $((_now-created)))
+		created_time=${created%.*}
+		edited_time=${edited%.*}
+		created_pretty=$(pretty_time $((_now-created_time)))
 		[ -n "${edited:-}" ] &&
-			edited_pretty=$(pretty_time $((_now-edited)))
+			edited_pretty=$(pretty_time $((_now-edited_time)))
 
 		set +eu
 		# Print with indentation
diff -Naur reddio-v0.46-orig/share/reddio/oauth.sh reddio-v0.46/share/reddio/oauth.sh
--- reddio-v0.46-orig/share/reddio/oauth.sh	2021-08-10 05:16:32.000000000 -0700
+++ reddio-v0.46/share/reddio/oauth.sh	2022-01-22 16:19:30.108566702 -0800
@@ -153,7 +153,9 @@
 
 	[ "$1" = "null" ] && {
 		access_token=$2
-		token_expires=$(($(date +%s) + $4))
+		_now=$(date +%s)
+		_now=${_now%.*}
+		token_expires=$((_now + $4))
 		refresh_token=$5
 
 		return 0
diff -Naur reddio-v0.46-orig/share/reddio/refresh-token.sh reddio-v0.46/share/reddio/refresh-token.sh
--- reddio-v0.46-orig/share/reddio/refresh-token.sh	2021-08-10 05:16:32.000000000 -0700
+++ reddio-v0.46/share/reddio/refresh-token.sh	2022-01-22 16:19:30.108566702 -0800
@@ -21,7 +21,9 @@
 
 	[ "$1" = "null" ] && {
 		access_token=$2
-		token_expires=$(($(date +%s) + $4))
+		_now=$(date +%s)
+		_now=${_now%.*}
+		token_expires=$((_now + $4))
 
 		debug 'Got new access token'
 
