diff -Naur sunwait-102cb417ecbb7a3757ba9ee4b94d6db3225124c4-orig/makefile sunwait-102cb417ecbb7a3757ba9ee4b94d6db3225124c4/makefile
--- sunwait-102cb417ecbb7a3757ba9ee4b94d6db3225124c4-orig/makefile	2020-10-26 13:39:53.000000000 -0700
+++ sunwait-102cb417ecbb7a3757ba9ee4b94d6db3225124c4/makefile	2021-02-12 20:02:59.471146883 -0800
@@ -8,7 +8,7 @@
 
 C=gcc
 CFLAGS=-c -Wall 
-LDFLAGS= -lm -lstdc++
+LDFLAGS+= -lm -lstdc++
 SOURCES=sunwait.cpp sunriset.cpp print.cpp
 HEADERS=sunwait.h sunriset.h print.h
 OBJECTS=$(SOURCES:.cpp=.o)
