diff -Naur securefs-orig/CMakeLists.txt securefs/CMakeLists.txt
--- securefs-orig/CMakeLists.txt	2020-08-20 23:52:45.370541605 -0700
+++ securefs/CMakeLists.txt	2020-08-21 01:22:12.772636903 -0700
@@ -19,13 +19,15 @@
 
 if (UNIX)
     find_path(FUSE_INCLUDE_DIR fuse.h PATHS /usr/local/include PATH_SUFFIXES osxfuse)
+    find_path(UTF8PROC_INCLUDE_DIR utf8proc.h PATHS /usr/local/include)
     if (APPLE)
         find_library(FUSE_LIBRARIES osxfuse PATHS /usr/local/lib)
     else()
         find_library(FUSE_LIBRARIES fuse PATHS /usr/local/lib)
+        find_library(UTF8PROC_LIBRARIES utf8proc PATHS /usr/local/lib)
     endif()
-    include_directories(${FUSE_INCLUDE_DIR})
-    link_libraries(${FUSE_LIBRARIES})
+    include_directories(${FUSE_INCLUDE_DIR} ${UTF8PROC_INCLUDE_DIR})
+    link_libraries(${FUSE_LIBRARIES} ${UTF8PROC_LIBRARIES})
     add_compile_options(-Wall -Wextra -Wno-unknown-pragmas)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --std=gnu++11")
 
@@ -72,11 +74,15 @@
 set(EXTERNAL_DIR external)
 include_directories(${EXTERNAL_DIR})
 
-set(CRYPTOPP_DIR ${EXTERNAL_DIR}/cryptopp)
-add_subdirectory(${CRYPTOPP_DIR})
-link_libraries(cryptopp-static)
+include(FindPkgConfig)
+pkg_check_modules(CRYPTOPP REQUIRED libcryptopp>=8.2)
+link_libraries(${CRYPTOPP_LIBRARIES})
+
+include(FindPkgConfig)
+pkg_check_modules(TCLAP REQUIRED tclap>=1.2.2)
+
 
-file(GLOB SOURCES sources/*.cpp sources/*.h ${EXTERNAL_DIR}/*.h ${EXTERNAL_DIR}/*.hpp ${EXTERNAL_DIR}/*.cpp ${CMAKE_BINARY_DIR}/git-version.cpp ${EXTERNAL_DIR}/utf8proc/utf8proc.h ${EXTERNAL_DIR}/utf8proc/utf8proc.c)
+file(GLOB SOURCES sources/*.cpp sources/*.h ${EXTERNAL_DIR}/*.hpp ${EXTERNAL_DIR}/*.cpp ${CMAKE_BINARY_DIR}/git-version.cpp)
 file(GLOB TEST_SOURCES test/*.cpp)
 add_library(securefs-static STATIC ${SOURCES})
 link_libraries(securefs-static)
diff -Naur securefs-orig/sources/commands.cpp securefs/sources/commands.cpp
--- securefs-orig/sources/commands.cpp	2020-08-20 23:52:45.406541543 -0700
+++ securefs/sources/commands.cpp	2020-08-21 00:40:20.980524299 -0700
@@ -14,7 +14,7 @@
 #include <fuse.h>
 #include <json/json.h>
 #include <tclap/CmdLine.h>
-#include <utf8proc/utf8proc.h>
+#include <utf8proc.h>
 
 #include <algorithm>
 #include <iostream>
diff -Naur securefs-orig/sources/mystring.cpp securefs/sources/mystring.cpp
--- securefs-orig/sources/mystring.cpp	2020-08-20 23:52:45.407541541 -0700
+++ securefs/sources/mystring.cpp	2020-08-21 00:40:20.981524308 -0700
@@ -3,7 +3,7 @@
 #include "logger.h"
 #include "myutils.h"
 
-#include <utf8proc/utf8proc.h>
+#include <utf8proc.h>
 
 #include <ctype.h>
 #include <errno.h>
