diff -Naur securefs-0.12.0-orig/CMakeLists.txt securefs-0.12.0/CMakeLists.txt
--- securefs-0.12.0-orig/CMakeLists.txt	2021-06-29 21:36:29.000000000 -0700
+++ securefs-0.12.0/CMakeLists.txt	2022-01-06 00:10:24.266965828 -0800
@@ -19,9 +19,13 @@
 
 if (UNIX)
     find_path(FUSE_INCLUDE_DIR fuse.h PATHS /usr/local/include)
+    find_path(UTF8PROC_INCLUDE_DIR utf8proc.h PATHS /usr/local/include)
     find_library(FUSE_LIBRARIES fuse PATHS /usr/local/lib)
+    find_library(UTF8PROC_LIBRARIES utf8proc PATHS /usr/local/lib)
     include_directories(${FUSE_INCLUDE_DIR})
     link_libraries(${FUSE_LIBRARIES})
+    include_directories(${FUSE_INCLUDE_DIR} ${UTF8PROC_INCLUDE_DIR})
+    link_libraries(${FUSE_LIBRARIES} ${UTF8PROC_LIBRARIES})
     add_compile_options(-Wall -Wextra -Wno-unknown-pragmas)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --std=gnu++11")
 
@@ -72,7 +76,7 @@
 add_subdirectory(${CRYPTOPP_DIR})
 link_libraries(cryptopp-static)
 
-file(GLOB SOURCES sources/*.cpp sources/*.h ${EXTERNAL_DIR}/*.h ${EXTERNAL_DIR}/*.hpp ${EXTERNAL_DIR}/*.cpp ${CMAKE_BINARY_DIR}/git-version.cpp ${EXTERNAL_DIR}/utf8proc/utf8proc.h ${EXTERNAL_DIR}/utf8proc/utf8proc.c)
+file(GLOB SOURCES sources/*.cpp sources/*.h ${EXTERNAL_DIR}/*.h ${EXTERNAL_DIR}/*.hpp ${EXTERNAL_DIR}/*.cpp ${CMAKE_BINARY_DIR}/git-version.cpp)
 file(GLOB TEST_SOURCES test/*.cpp)
 add_library(securefs-static STATIC ${SOURCES})
 link_libraries(securefs-static)
diff -Naur securefs-0.12.0-orig/sources/commands.cpp securefs-0.12.0/sources/commands.cpp
--- securefs-0.12.0-orig/sources/commands.cpp	2021-06-29 21:36:29.000000000 -0700
+++ securefs-0.12.0/sources/commands.cpp	2022-01-05 23:28:36.519567975 -0800
@@ -14,7 +14,7 @@
 #include <fuse.h>
 #include <json/json.h>
 #include <tclap/CmdLine.h>
-#include <utf8proc/utf8proc.h>
+#include <utf8proc.h>
 
 #include <algorithm>
 #include <iostream>
diff -Naur securefs-0.12.0-orig/sources/mystring.cpp securefs-0.12.0/sources/mystring.cpp
--- securefs-0.12.0-orig/sources/mystring.cpp	2021-06-29 21:36:29.000000000 -0700
+++ securefs-0.12.0/sources/mystring.cpp	2022-01-05 23:28:36.519567975 -0800
@@ -3,7 +3,7 @@
 #include "logger.h"
 #include "myutils.h"
 
-#include <utf8proc/utf8proc.h>
+#include <utf8proc.h>
 
 #include <ctype.h>
 #include <errno.h>
