diff -Naur securefs-0.12.0-orig/external/catch.hpp securefs-0.12.0/external/catch.hpp
--- securefs-0.12.0-orig/external/catch.hpp	2021-06-29 21:36:29.000000000 -0700
+++ securefs-0.12.0/external/catch.hpp	2022-01-06 00:03:30.737400869 -0800
@@ -6458,13 +6458,15 @@
             { SIGTERM, "SIGTERM - Termination request signal" },
             { SIGABRT, "SIGABRT - Abort (abnormal termination) signal" }
     };
+	constexpr std::size_t minStackSizeForErrors = 32 * 1024;
+    static char* altStackMem = nullptr;
+    static std::size_t altStackSize = 0;
 
     struct FatalConditionHandler {
 
         static bool isSet;
         static struct sigaction oldSigActions [sizeof(signalDefs)/sizeof(SignalDefs)];
         static stack_t oldSigStack;
-        static char altStackMem[SIGSTKSZ];
 
         static void handleSignal( int sig ) {
             std::string name = "<unknown signal>";
@@ -6481,6 +6483,10 @@
         }
 
         FatalConditionHandler() {
+			if (altStackSize == 0) {
+				altStackSize = std::max(static_cast<size_t>(SIGSTKSZ), minStackSizeForErrors);
+			}
+			altStackMem = new char[altStackSize]();
             isSet = true;
             stack_t sigStack;
             sigStack.ss_sp = altStackMem;
@@ -6498,6 +6504,8 @@
 
         ~FatalConditionHandler() {
             reset();
+			delete[] altStackMem;
+			altStackMem = nullptr;
         }
         static void reset() {
             if( isSet ) {
@@ -6515,7 +6523,6 @@
     bool FatalConditionHandler::isSet = false;
     struct sigaction FatalConditionHandler::oldSigActions[sizeof(signalDefs)/sizeof(SignalDefs)] = {};
     stack_t FatalConditionHandler::oldSigStack = {};
-    char FatalConditionHandler::altStackMem[SIGSTKSZ] = {};
 
 } // namespace Catch
 
