diff -Naur python-slugify-4.0.0-orig/setup.py python-slugify-4.0.0/setup.py
--- python-slugify-4.0.0-orig/setup.py	2019-10-20 10:39:51.000000000 -0700
+++ python-slugify-4.0.0/setup.py	2020-05-17 13:54:51.678426207 -0700
@@ -14,8 +14,8 @@
 author = 'Val Neekman'
 author_email = 'info@neekware.com'
 license = 'MIT'
-install_requires = ['text-unidecode>=1.3']
-extras_require = {'unidecode': ['Unidecode>=1.1.1']}
+install_requires = ['Unidecode>=1.1.1']
+extras_require = {}
 
 classifiers = [
     'Development Status :: 5 - Production/Stable',
